examples/client/lwm2mclient.c

```diff
@@ -118,14 +126,30 @@ static void prv_quit(lwm2m_context_t * lwm2mH,
     (void)buffer;
     (void)user_data;
 
+    // Set the quit flag to terminate the program
     g_quit = 1;
 }
 
+/**
+ * @brief Signal handler for SIGINT (Ctrl+C) for graceful shutdown.
+ *
+ * @param lwm2mH LwM2M context.
+ * @param uri URI of the resource.
+ */
 void handle_sigint(int signum)
 {
+    // Set the quit flag to initiate a graceful shutdown
     g_quit = 2;
 }
 
+/**
+ * @brief Callback function for handling value changes in resources.
+ *
+ * @param lwm2mH LwM2M context.
+ * @param uri URI of the resource.
+ * @param value New value of the resource.
+ * @param valueLength Length of the new value.
+ */
 void handle_value_changed(lwm2m_context_t * lwm2mH,
                           lwm2m_uri_t * uri,
                           const char * value,
```

Comment:
also we should add a definition reference or a comment about what does these definition and function do. E.g. `LWM2M_LIST_FIND` is defined as: https://github.com/eniqo-energy/wakaama/blob/9b09fac16a855bf9be908d6e0094b6611703daef/core/list.c#L47

E.g. of a comment should be: `@brief Iterate function for finding element in a list of the lwm2m_list_t pointer by id`